bin/cli.js

```diff
@@ -1,17 +1,255 @@
+  'Firefox':   'karma-firefox-launcher',
+  'IE':        'karma-ie-launcher',
+  'Opera':     'karma-opera-launcher',
+  'PhantomJS': 'karma-phantomjs-launcher',
+  'Safari':    'karma-safari-launcher'
+};
+
+const merge = (arr1 = [], arr2 = []) => common.merge(arr1, arr2);
+
+const exclude = (files, filterArr) =>
+  filterArr.map(path => path.replace('.', '\\.')
+    .replace('*', '.+')
+    .replace('?', '.'))
+    .map(path => new RegExp(path))
+    .reduce((files, regexp) => files.filter(file => !regexp.test(file)), files);
 
-walk('../test', (err, files) => {
-  metasync.for(files).each((file, next) => {
-    require(file.filePath);
-    next();
-  }).fetch(() => {
-    console.log('Tests done');
+const parseFile = file => {
+  const data = fs.readFileSync(path.resolve(file), 'utf8');
+  switch (common.fileExt(file)) {
+    case 'json':
+      return JSON.parse(data);
+    case 'yml':
+    case 'yaml':
+      return yaml.parse(data);
+    default:
+      return {};
+  }
+};
+
+const loadFiles = files => {
+  const result = [];
+  files
+    .map(file => {
+      if (fs.existsSync(file + '.js')) {
+        return file + '.js';
+      } else if (fs.existsSync(file)) {
+        return file;
+      } else {
+        console.error('File does not exist:', file);
+        process.exit(1);
+      }
+    })
+    .forEach(file => {
+      if (fs.statSync(file).isDirectory()) {
+        const subfiles = fs.readdirSync(file).map(f => `${file}/${f}`);
+        result.push(...loadFiles(subfiles));
+      } else if (path.extname(file) === '.js') {
+        result.push(file);
+      }
+    });
+  return result;
+};
+
+const setKarmaBrowsers = (config, ...browsers) => {
+  if (!config.plugins) config.plugins = [];
+  if (!config.browsers) config.browsers = [];
+
+  browsers.forEach(browser => {
+    const launcher = browserLaunchers[browser];
+    if (!launcher) {
+      console.error('Metatests library does not support such browser:',
+        browser);
+      process.exit(1);
+    }
+    config.browsers.push(browser);
+    config.plugins.push(launcher);
   });
-});
+};
+
+const setKarmaLogLevel = (config, logLevel = 'disable') => {
+  if (!config.webpackMiddleware) config.webpackMiddleware = {};
+  config.logLevel = karma.constants[karmaLogLevels[logLevel][0]];
+  config.webpackMiddleware.stats = karmaLogLevels[logLevel][1];
+};
+
+const removeNodePackages = (config, ...packages) => {
+  if (!config.webpack) config.webpack = {};
+  if (!config.webpack.node) config.webpack.node = {};
+  packages.forEach(lib => config.webpack.node[lib] = 'empty');
+};
+
+const getReporter = () => {
+  const browsers = [];
+  const reporter = function(base) {
+    base(this);
+    this.onBrowserLog = (browser, log) => {
+      if (typeof log !== 'string') log = JSON.stringify(log);
+      if (!browsers.includes(browser)) {
+        browsers.push(browser);
+        console.log(`\n${browser}:`);
+      }
+      console.log(log.slice(1, log.length - 1));
```

Comment:
Because log already has quotes e.g. `\'something\'` and I didn't find another way of preventing them added except changing internals of karma. Removing them is not necessary but it will mess up output a bit.